### DISCUSSION TITLE
how to reference a page inside the data grid ?? also how to reference a local file from the grid

### DISCUSSION THREAD

**Discussion by ameerfayiz** [AppFlowy-IO/AppFlowy#3619] [General help wanted]
i have a database grid , i want ot reference other documents in it so that i can go to that page by clicking that link in database grid. also i have to  link local file from the databse grid so that i can open local file using that link . I tried  using ```file://C:/Users/Ameer/Downloads/file.wav``` but what happens is , the link is opened by stripping the colun character after the ```file``` part.  so the link cannot be opened , that is, link is opened as ```file//C:/Users/Ameer/Downloads/file.wav```

**Comment by Digital365Staking** [comment]
It seems that AppFlowy's database grid feature is still evolving, and linking to external documents and local files may not yet be fully refined. Let's break down both of your needs and possible workarounds:

### 1. **Linking to Other Documents within AppFlowy**
For referencing other documents within your AppFlowy database grid:
- You can create links to other pages using internal page URLs (if AppFlowy supports this functionality) by copying the page link and pasting it into the grid's relevant field. This would make the page clickable within the database grid. 
- If AppFlowy doesn't support this directly, you may need to check if internal page linking is available in the roadmap or explore alternative ways to organize the content, such as adding navigable tags or manually including links in text fields.

### 2. **Linking to Local Files**
For linking local files (e.g., `file://C:/Users/Ameer/Downloads/file.wav`), the issue you're encountering likely arises from the URL encoding or how AppFlowy parses URLs.

The stripped colon (resulting in `file//C:/...`) suggests that the `file://` URL scheme is not fully supported or is being misinterpreted. Here are a couple of workarounds:

#### **Workaround 1: Encode the URL**
Try URL-encoding the link by replacing special characters, such as:
- Replace the colon (`:`) with `%3A`
- The file URL would become: `file%3A//C:/Users/Ameer/Downloads/file.wav`

#### **Workaround 2: Use a Shortcut**
You could create a shortcut (or symbolic link) to the local file and place it in a directory where AppFlowy might allow regular linking (like a cloud storage directory, which can be linked through `https://`). Then use the web-accessible link instead of `file://`.

#### **Workaround 3: Explore Integrations**
If AppFlowy eventually supports file or folder integration with services like OneDrive, Google Drive, or Dropbox, it may be easier to reference files stored in those services via `https` links, instead of local file paths.

Until these features are fully implemented, I recommend checking with the AppFlowy community or GitHub page for updates or workarounds specific to AppFlowy's development timeline.
